### PR TITLE
Add optics-compat for compatibility layer between van Laarhoven optics

### LINKED ISSUE #165: Add isoVL and prismVL

### PR DESCRIPTION
**Pull request by arybczak**
Fixes #165.

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -1,5 +1,6 @@
 packages:
   optics/*.cabal
+  optics-compat/*.cabal
```
Comment (adamgundry) [inline cabal.project]: The CI will need updating, modulo #181.

File: optics-compat/optics-compat.cabal [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+description:
+  This package provides utils for compatibility with van Laarhoven optics
+
+library
+  default-language: Haskell2010
+  hs-source-dirs:   src
+  ghc-options:      -Wall
+  build-depends: base          >=4.9     && <5
+               , optics-core   >=0.1     && <1
+               , profunctors   >=5.0     && <6.0
```
Comment (adamgundry) [inline optics-compat/optics-compat.cabal]: Presumably we'll leave `optics-compat` out of `optics` so as not to impose a dependency on `profunctors`?

**Comment by phadej** [comment]
`optics-compat` is not a good name. How about `optics-van-laarhoven` or `optics-vl`?

**Comment by adamgundry** [comment]
I'd wondered about the name too. I have a weak preference for `optics-vl`.

**Comment by arybczak** [comment]
Yeah, `optics-vl` sounds good.

File: optics-compat/src/Optics/Compat.hs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  (
+  -- * Iso
+    IsoVL
+  , IsoVL'
+  , isoVL
+  -- * Prism
+  , PrismVL
+  , PrismVL'
+  , prismVL
+  ) where
```
[the remark this answers is not in the document]
Comment (arybczak) [reply optics-compat/src/Optics/Compat.hs]: Done.

File: optics-compat/optics-compat.cabal (hunk 2 above)
Comment (arybczak) [reply optics-compat/optics-compat.cabal]: Yes.

**Comment by arybczak** [comment]
I renamed `optics-compat` to `optics-vl` and included all `VL` stuff in the `Optics.VL` module.

File: cabal.project (hunk 1 above)
Comment (phadej) [reply cabal.project]: #181 is merged.

**Comment by arybczak** [comment]
I updated `travis.yml`. Do we want `ToReadOnly` instances for read-only optic kinds?

**Comment by arybczak** [comment]
I think it's fine to merge it now.

File: optics-vl/LICENSE [hunk 4]
```diff
@@ -0,0 +1,30 @@
+Copyright (c) 2017-2018, Well-Typed LLP
```
Comment (phadej) [inline optics-vl/LICENSE]: we'll need to check years in some later sweep-scan.

**Pull request merged**